src/data/markdown/docs/05 Examples/01 Examples/05 data-parameterization.md

```diff
@@ -16,14 +16,21 @@ This will, in turn, make your test more realistic.
 
 <Blockquote mod="warning">
 
-#### ⚠️ Strive to keep the data files small
-
-Each VU in k6 will have its separate copy of the data file.
-If your script uses 300 VUs, there will be 300 copies of the data file in memory.
-Cloud service allots 8GB of memory for every 300VUs.
-When executing cloud tests, make sure your data files aren't exceeding this limit or your test run may get aborted.
-
-Starting with k6 v0.27.0, there are some [tricks that can be used to better handle bigger data files](#handling-bigger-data-files).
+Each VU in k6 is a separate JS VM so in order to not have multiple copies of the whole data file
+[SharedArray](/javascript-api/k6-data/sharedarray) was added. It does have overhead in accessing elements compared to a normal non shared
+array, but the difference is negligible compared to the time it takes to make requests. And is
+especially not comparable as not using it with big files basically means k6 will be using too much
+memory to run. Which might lead to your script not being able to run at all or aborting in the
+middle.
+
+For example, the Cloud service allots 8GB of memory for every 300VUs. So if you are files are big
+enough and you are not using [SharedArray](/javascript-api/k6-data/sharedarray), that might mean that your script will run out of memory at
+some point. Additionally even if there is enough memory, k6 has a garbage collector(as it's written
+in golang) and it will walk through all accessible objects (including JS ones) and figure out which
+need to be garbage collected. For big JS arrays copied hundreds of times this adds quite a lot of
+additional work.
```

Comment:
The second sentence just repeats what the last sentence of the previous paragraph said, and mentioning the garbage collector seems like too much detail for this documentation.

Mentioning the Cloud limits might be relevant, but we'll have to keep it updated if that changes, and I'm not sure if it's worth bringing it up at all or if this is the right place for it. So I would scrap this paragraph entirely, but if not some minor suggestions:

```suggestion
For example, the Cloud service allocates 8GB of memory for every 300 VUs. So if your files are large
enough and you are not using [SharedArray](/javascript-api/k6-data/sharedarray), that might mean that your script will run out of memory at
some point. Additionally even if there is enough memory, k6 has a garbage collector (as it's written
in golang) and it will walk through all accessible objects (including JS ones) and figure out which
need to be garbage collected. For big JS arrays copied hundreds of times this adds quite a lot of
additional work.
```

Reply:
The cloud limits were here from before. I am not certain this is the correct place for them as well, but it is definetely good to have them somewhere. Given that if a cloud user runs out of memory they might be directed to this page, I am for this actually staying. It also shows how much memory we have decided to allocate to a given number of VUs.

As to change ... things changes, so we will need to update it if we ever  change it in the cloud :)